packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2941,9 +2944,39 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           this.resetStartLocationFromNode(expr, typeParameters);
         }
         expr.typeParameters = typeParameters;
+
+        // report error if single type parameter used without trailing comma.
+        if (
+          this.hasPlugin("jsx") &&
+          expr.typeParameters.params.length === 1 &&
+          !expr.typeParameters.extra?.trailingComma
+        ) {
+          const parameter = expr.typeParameters.params[0];
+          if (parameter.constraint) {
+            // If parameter has any constraints, it must contain multiple tokens.
+            // <T extends U> is a valid declaration.
+            // <T extends {name: string}> is also a valid declaration.
+          } else invalidSingleType = parameter;
+        }
+
         return expr;
       }, state);
 
+      if (invalidSingleType) {
+        this.raise(
+          TSErrors.SingleTypeParameterWithoutTrailingComma,
+          {
+            at: {
+              ...invalidSingleType.loc.end,
+              column: invalidSingleType.loc.end.column + 1,
```

Comment:
There is `createPositionWithColumnOffset` in `./util/location` serving the same purpose.

Reply:
Oh I didn't know that. So you want me to use that method for the position and update the PR right?
@JLHwung

Reply:
Yes, please!